applications/tari_base_node/src/commands/command/mod.rs

```diff
@@ -193,8 +193,50 @@ impl CommandContext {
         if let Command::Watch(command) = args.command {
             Ok(Some(command))
         } else {
+            let time_out = match args.command {
+                // These commands should complete quickly, some of them like 'discover-peer' returns immediately
+                // although the requested action can take a long time
+                Command::Version(_) |
+                Command::Whoami(_) |
+                Command::CheckForUpdates(_) |
+                Command::AddPeer(_) |
+                Command::BanPeer(_) |
+                Command::UnbanAllPeers(_) |
+                Command::UnbanPeer(_) |
+                Command::GetPeer(_) |
+                Command::ResetOfflinePeers(_) |
+                Command::DialPeer(_) |
+                Command::PingPeer(_) |
+                Command::DiscoverPeer(_) |
+                Command::ListPeers(_) |
+                Command::ListBannedPeers(_) |
+                Command::ListConnections(_) |
+                Command::GetNetworkStats(_) |
+                Command::BlockTiming(_) |
+                Command::GetChainMetadata(_) |
+                Command::GetDbStats(_) |
+                Command::GetStateInfo(_) |
+                Command::ListReorgs(_) |
+                Command::GetBlock(_) |
+                Command::ListHeaders(_) |
+                Command::HeaderStats(_) |
+                Command::SearchUtxo(_) |
+                Command::SearchKernel(_) |
+                Command::Quit(_) |
+                Command::Exit(_) => 30,
+                // Depending on the state and busyness of the mempool, these commands can take a fairly long time
+                Command::Status(_) |
+                Command::Watch(_) |
+                Command::GetMempoolStats(_) |
+                Command::GetMempoolState(_) |
+                Command::GetMempoolTx(_) => 180,
```

Comment:
I would make the mempool and status much shorter, since they should complete quickly. It's annoying to have to wait 70 seconds (nevermind 180) for it to complete before you can use the prompt again. I'd say make it 5 seconds.

Reply:
Ideally, we should handle a ctrl+c signal and abort the command without any timeouts

Reply:
My train of thought was to at least display something. With my testing, even 180 s is sometimes not enough. Under normal circumstances, it will return quickly and by the off chance that a mempool query takes longer sufficient time is allowed.

Reply:
I proposed something that could solve the problem, please see the latest commit.